pkg/odo/cli/component/common_link.go

```diff
@@ -128,20 +128,18 @@ func (o *commonLinkOptions) complete(name string, cmd *cobra.Command, args []str
 		// make this deployment the owner of the link we're creating so that link gets deleted upon doing "odo delete"
 		ownerReference := generator.GetOwnerReference(deployment)
 		o.serviceBinding.SetOwnerReferences(append(o.serviceBinding.GetOwnerReferences(), ownerReference))
+
+		deploymentGVR, err := o.KClient.GetDeploymentAPIVersion()
 		if err != nil {
 			return err
 		}
 
-		// This is a really hacky way to get group, version and resource info but I couldn't find better one.
-		// A sample "deploymentSelfLinkSplit" looks like: [ apis apps v1 namespaces myproject deployments nodejs ]
-		deploymentSelfLinkSplit := strings.Split(deployment.SelfLink, "/")
-
 		// Populate the application selector field in service binding request
 		o.serviceBinding.Spec.Application = &servicebinding.Application{
 			GroupVersionResource: metav1.GroupVersionResource{
```

Comment:
the original reason why I suggested using `metav1.GroupVersionResource` as the return type for `GetDeploymentAPIVersion`  is that you could use it here directly  like this
```
GroupVersionResource: deploymentGVR
```